lib/strong_resources/controller/mixin.rb

```diff
@@ -21,10 +22,24 @@ def update_action?
         action_name == 'update'
       end
 
+      private
+
+      def _params_type
+        params_method = if respond_to?(:raw_params) 
+                          :raw_params 
+                        else
+                          :params
+                        end
+
+        send(params_method).try(:[], :data).try(:[], :type).try(:to_sym)
+      end
+
       module ClassMethods
         def strong_resource(name, opts = {}, &blk)
+          blk ||= Proc.new {}
           opts[:require] ||= name unless opts[:require] == false
           resource = StrongResource.from(name, opts, &blk)
+          type = opts[:type] || name
```

Comment:
could we make this `jsonapi_type`? Also, shouldn't this be the pluralized name? I think there is already code somewhere taking the pluralized name...